docs(keyring) additional notes regarding keyring behavior (#1692)
Kong/docs.konghq.com: app/enterprise/1.3-x/db-encryption.md

All nodes in the Kong cluster should share the same `keyring_enabled` and `keyring_strategy` configuration values. Not every node needs to be provided the management RSA key pair, as that key pair is only used for backup and recovery processes. It does not need to be present for regular database read/write operations.

> Note that the user under which Kong worker processes run (as defined by the `nginx_user` Kong configuration option) must have read access to the public and private keys in order to be able to perform keyring export and import operations. We recommend restricting access to these files as tightly as possible. For example:

Reviewer: ```suggestion
The Kong worker processes run under the user you define in Kong's configuration property, `nginx_user`. Note that this user must have read access to the public and private keys in order to be able to perform keyring export and import operations. We recommend restricting access to these files as tightly as possible. For example:
```

Author: to me this reads oddly that we start the sentence discussing the kong user, with no context, followed by a hard stop. this is one thought- "the user must have read access..." vs "there is a user**.** it must have read access..."

will defer to you here

Reviewer: I agree that it should have some context. I wanted to break apart the sentence since I found it hard to follow. Do you think this conveys enough detail?

```suggestion
Note that the user (defined by the `nginx_user` property) must have read access to the public and private keys in order to be able to perform keyring export and import operations. We recommend restricting access to these files as tightly as possible. For example:
```

> Note that the user under which Kong worker processes run (as defined by the `nginx_user` Kong configuration option) must have read access to the public and private keys in order to be able to perform keyring export and import operations. We recommend restricting access to these files as tightly as possible. For example:

Author: ```suggestion
Note that the user under which Kong worker processes run must have read access to the public and private keys in order to be able to perform keyring export and import operations. This user is defined by the `nginx_user` Kong configuration option. We recommend restricting access to these files as tightly as possible. For example:
```

Author: @h3xar0n alternate suggestion? ^

Reviewer: :shipit:

For all other nodes, the generated key will be automatically distributed within a few seconds.

> Note that encryption keys are held _only_ in memory, and will not persist past a restart of the Kong process (e.g., running `kong restart`). Because of this, you must export the keyring following it's initialization, or any sensitive fields written with the keyring will not be recoverable if all Kong nodes in a cluster restart simultaneously. Key material _will_ persist after a soft reload of Kong (e.g., `kong reload`).

Reviewer: ```suggestion
Note that encryption keys are held _only_ in memory, and do not persist past a restart of the Kong process (e.g., running `kong restart`). Because of this limitation, you must export the keyring following its initialization. Otherwise, if all Kong nodes in a cluster restart simultaneously, any sensitive fields written with the keyring become unrecoverable. Key material still persists after a soft reload of Kong (i.e., `kong reload`).
```

When operating in `cluster` mode, keyring material propagates automatically among all nodes in the Kong cluster. Because Kong nodes do not have a notion of direct peer-to-peer communication, the underlying datastore (Cassandra or PostgreSQL) serves as a communication channel to transmit messages. When a Kong node starts, it generates an ephemeral RSA key pair. The node's public keys propagate to all other active nodes in the cluster. When an active node sees a message request for keyring material, it wraps the in-memory keyring material in the presented public key, and transmits the payload back over the central messaging channel provided by the underlying data store. This process allows each node in the cluster to broadcast keyring material to new nodes, without sending key material in plaintext over the wire. This model requires that at least one node be running at all times within the cluster; a failure of all nodes requires manually re-importing the keyring to one node during an outage recovery.

### Encrypted Fields

> The following fields are encrypted as rest by the keyring module:

Reviewer: ```suggestion
The keyring module encrypts the following fields at rest:
```
